lib/AST/Decl.cpp

```diff
@@ -3729,6 +3730,10 @@ Type EnumDecl::getRawType() const {
                        TypeResolutionStage::Interface}, Type());
 }
 
+bool EnumDecl::isCXXNamespace() const {
```

Comment:
Hm, this seems like the sort of hardcoding that doesn't make sense to expose to the rest of the compiler. What do you think about just assuming that enums with Clang nodes and no cases are always namespacey?

Reply:
I was worried about the following case, but I can't seem to trigger the code in question.
```c++
enum Blah : char {};
Blah impossible() { char silly = 3; return *reinterpret_cast<Blah*>(&silly); }
```
```swift
print(type(of: impossible())) // should print: Int8
print(impossible()) // should print: Blah(rawValue: 3)
```

Reply:
Hm, good point. Try

```cxx
enum __attribute__((enum_extensibility(open))) Blah : char {};
Blah impossible() { char silly = 3; return *reinterpret_cast<Blah*>(&silly); }
```

Any setting for `enum_extensibility` will force Swift to import it as a true enum unless it also has the `flag_enum` attribute.

Maybe the right answer here is to check for `@objc` first (and make sure we set that consistently on imported enums), and then fall back to this second.

Reply:
Hmm, curious. When you put enum_extensibility on it, swift is more strict and treats it as actually being impossible. The generated code then segfaults because of undefined behavior. In the linux case, it is not going to trigger that @objc check. It looks highly unlikely that anyone relies on this.

Reply:
Hrm. That's a bug; imported C enums can always have inhabitants beyond what's officially declared. We might not have to fix it right away, but I think we shouldn't make it *worse.*

Reply:
Alright, with the !isObjC check, these problem enums no longer trigger this special c++-namespace handling enums. They do, however, trigger the same compiler crasher that I was seeing before with namespaces, so I suspect that whatever would fix things for these problem enums would fix it for me as well.